File: leetcode/169. Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int> m;
        int size = nums.size();
        for(int i = 0 ; i < size ; i++){
            m[nums[i]]++;
            if(m[nums[i]] > size/2) return nums[i];
        }
        return -1;
    }
};

File: online-training-platform/leetcode/169. Majority Element.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        val limit = nums.size / 2
        var counter = 1

        val sorted = nums.sorted()
        var num = sorted.first()
        for(idx in 1 until sorted.size) {
            if(sorted[idx] == num){
                counter += 1
            } else {
                num = sorted[idx]
                counter = 1
            }

            if(counter > limit){
                return sorted[idx]
            }
        }

        return num
    }
}